include/boost/url/grammar/sequence_rule.hpp

```diff
@@ -130,6 +131,98 @@ sequence_rule(
 }
 #endif
 
+#ifndef BOOST_URL_DOCS
+namespace detail {
+
+template<class Rule>
+struct squelch_rule_t
+    : urls::detail::empty_value<Rule>
+{
+    using value_type = void;
+
+    constexpr
+    squelch_rule_t(
+        Rule const& r) noexcept
+        : urls::detail::empty_value<Rule>(
+            urls::detail::empty_init, r)
+    {
+    }
+
+    result<value_type>
+    parse(
+        char const*& it,
+        char const* end) const
+    {
+        auto rv = this->get().parse(it, end);
+        if(rv.error())
+            return rv.error();
+        return {}; // void
+    }
+};
+
+} // detail
+#endif
+
+/** Squelch the value of a rule
+
+    This function returns a new rule which
+    matches the specified rule, and converts
+    its value type to `void`. This is useful
+    for matching delimiters in a grammar,
+    where the value for the delimiter is not
+    needed.
+
+    @par Value Type
+    @code
+    using value_type = void;
+    @endcode
+
+    @par Example 1
+    With `squelch`:
+    @code
+    result< std::tuple< pct_encoded_view, string_view > > rv = grammar::parse(
+        "www.example.com:443",
+        grammar::sequence_rule(
+            pct_encoded_rule(unreserved_chars + '-' + '.'),
+            grammar::squelch( grammar::char_rule( ':' ) ),
+            grammar::token_rule( grammar::digit_chars ) ) );
+    @endcode
+
+    @par Example 2
+    Without `squelch`:
+    @code
+    result< std::tuple< pct_encoded_view, string_view, string_view > > rv = grammar::parse(
+        "www.example.com:443",
+        grammar::sequence_rule(
+            pct_encoded_rule(unreserved_chars + '-' + '.'),
+            grammar::char_rule( ':' ),
+            grammar::token_rule( grammar::digit_chars ) ) );
+    @endcode
+
+    @par r The rule to squelch
+
+    @see
+        @ref grammar::char_rule,
+        @ref grammar::digit_chars,
+        @ref grammar::parse,
+        @ref grammar::sequence,
+        @ref grammar::token_rule,
+        @ref pct_encoded_view,
+        @ref pct_encoded_rule,
+        @ref unreserved_chars.
+*/
+template<class Rule>
+constexpr
+#ifdef BOOST_URL_DOCS
+__implementation_defined__
+#else
+detail::squelch_rule_t<Rule>
+#endif
+squelch( Rule const& r ) noexcept
```

Comment:
Oh... not a detail. I think I would just call it "ignore_value" or something.

```cpp
result< std::tuple< pct_encoded_view, string_view > > rv = grammar::parse(
        "www.example.com:443",
        grammar::sequence_rule(
            pct_encoded_rule(unreserved_chars + '-' + '.'),
            grammar::ignore_value( grammar::char_rule( ':' ) ),
            grammar::token_rule( grammar::digit_chars ) ) );
```

Reply:
I like `squelch` over `ignore_value` since squelch is shorter.

Reply:
`squelch` took me a long time because I had to look in the dictionary. :laughing:

Reply:
Nothing wrong with that in fact, its a bonus because now we have another tool in the toolbox (a new word).